Touch Bar Simulator/TouchBarWindow.swift

```diff
@@ -45,6 +66,177 @@ final class TouchBarWindow: NSPanel {
+	}
+
+	func stopDockBehaviorTimer() {
+		dockBehaviorTimer.invalidate()
+		dockBehaviorTimer = Timer()
+	}
+
+	var dockBehavior: Bool = defaults[.dockBehavior] {
+		didSet {
+			guard self.docking != nil else {
+				return
+			}
+
+			if self.docking! == .dockedToBottom || self.docking! == .dockedToTop {
+				defaults[.lastWindowDockingWithDockBehavior] = self.docking!
+			}
+
+			if dockBehavior {
+				if self.docking! == .dockedToBottom || self.docking! == .dockedToTop {
+					self.docking = defaults[.lastWindowDockingWithDockBehavior]
+					startDockBehaviorTimer()
+				} else if self.docking! == .floating {
+					defaults[.windowDocking] = defaults[.lastWindowDockingWithDockBehavior]
+				}
+			} else {
+				stopDockBehaviorTimer()
+				self.setIsVisible(true)
+			}
+		}
+	}
+
+	@objc
+	func handleDockBehavior() {
+		guard
+			self.docking != nil,
+			let visibleFrame = NSScreen.main?.visibleFrame,
+			let screenFrame = NSScreen.main?.frame
+		else {
+			return
+		}
+
+		var detectionRect: NSRect = .zero
+		if self.docking! == .dockedToBottom {
+			if self.isVisible {
+				detectionRect = CGRect(
+					x: 0,
+					y: 0,
+					width: visibleFrame.width,
+					height: self.frame.height + (screenFrame.height - visibleFrame.height - NSStatusBar.system.thickness)
+				)
+			} else {
+				detectionRect = CGRect(x: 0, y: 0, width: visibleFrame.width, height: 1)
+			}
+		} else if self.docking! == .dockedToTop {
+			if self.isVisible {
+				detectionRect = CGRect(
+					x: 0,
+					// Without `+ 1`, the Touch Bar would glitch (toggling rapidly).
+					y: screenFrame.height - self.frame.height - NSStatusBar.system.thickness + 1,
+					width: visibleFrame.width,
+					height: self.frame.height + NSStatusBar.system.thickness)
+			} else {
+				detectionRect = CGRect(
+					x: 0,
+					y: screenFrame.height,
+					width: visibleFrame.width,
+					height: 1)
+			}
+		}
+
+		let mouseLocation = NSEvent.mouseLocation
+		if detectionRect.contains(mouseLocation) {
+			dismissAnimationDidRun = false
+
+			guard
+				!showTouchBarTimer.isValid,
+				!showAnimationDidRun
+			else {
+				return
+			}
+
+			showTouchBarTimer = Timer.scheduledTimer(withTimeInterval: 0.3, repeats: false) { _ in
+				self.showTouchBarWithAnimation()
+				self.showAnimationDidRun = true
+			}
+		} else {
+			showTouchBarTimer.invalidate()
+			showTouchBarTimer = Timer()
+			showAnimationDidRun = false
+
+			if self.isVisible && !dismissAnimationDidRun {
+				dismissTouchBarWithAnimation()
+				dismissAnimationDidRun = true
+			}
+		}
+	}
+
+	func moveToStartPoint() {
+		guard self.docking != nil else {
+			return
+		}
+
+		if self.docking! == .dockedToTop {
+			self.moveTo(x: .center, y: .top)
+		} else if self.docking! == .dockedToBottom {
+			self.moveTo(x: .center, y: .bottom)
+		}
+	}
+
+	var showAnimationDidRun = false
+	var dismissAnimationDidRun = false
+
+	func showTouchBarWithAnimation() {
```

Comment:
There's a lot of duplicated code between this method and `dismissTouchBarWithAnimation`.